### PR TITLE
Renamed KeyPoint to Keypoint

### PR DESCRIPTION
**Pull request by mkamalza**
# Peer Review Information:
Information on any code, feature, documentation changes here

- Removed capitalized Points across the repo

## Editor / Package versioning:
**Editor Version Target (i.e. 19.3, 20.1)**: 2019.3

## Dev Testing:
**Tests Added**: 

**Core Scenario Tested**: 

**At Risk Areas**: 

**Notes + Expectations**: 

## Checklist
- [ ] - Updated docs
- [ ] - Updated changelog

### CODE REVIEW

**Review by JonathanHUnity — CHANGES_REQUESTED** [review]
Great! Don't forget the change log!

File: com.unity.perception/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 
+Renamed all instances of "KeyPoint" to "Keypoint" for standards adherence.
```
Comment (mkamalza) [inline com.unity.perception/CHANGELOG.md:18]: @JonathanHUnity I was having a hard time coming up with a good log line here. Please feel free to change this!

Comment (jhogins) [reply com.unity.perception/CHANGELOG.md:18]:
```suggestion
Renamed all types and members named "KeyPoint" to "Keypoint"
```
How about something like this?

File: com.unity.perception/Runtime/GroundTruth/Labelers/KeyPointLabeler.cs [hunk 2]
```diff
@@ -79,14 +79,14 @@ public KeyPointLabeler(IdLabelConfig config, KeyPointTemplate template)
         protected override void Setup()
         {
             if (idLabelConfig == null)
-                throw new InvalidOperationException("KeyPointLabeler's idLabelConfig field must be assigned");
+                throw new InvalidOperationException($"{GetType().Name}'s idLabelConfig field must be assigned");
```
Comment (jhogins) [inline com.unity.perception/Runtime/GroundTruth/Labelers/KeyPointLabeler.cs:82]: Why this change? `nameof(KeypointLabeler)` is more appropriate if you don't expect inheritance.

File: com.unity.perception/CHANGELOG.md (hunk 1 above)
Comment (jhogins) [reply com.unity.perception/CHANGELOG.md:18]: Also, we should add an upgrade note as well on changes they will have to make. It's going to be a pain for any current users.

**Review by jhogins — CHANGES_REQUESTED** [review]
Couple minor changes I'd suggest

**Review by JonathanHUnity — CHANGES_REQUESTED** [review]
Who is that @jhogins imposter :D

File: com.unity.perception/Runtime/GroundTruth/Labelers/KeyPointLabeler.cs (hunk 2 above)
Comment (mkamalza) [reply com.unity.perception/Runtime/GroundTruth/Labelers/KeyPointLabeler.cs:82]: Good point, I'll change that.

File: com.unity.perception/CHANGELOG.md (hunk 1 above)
Comment (mkamalza) [reply com.unity.perception/CHANGELOG.md]: > How about something like this?

I like it. But technically, it is also other appearances of the term as part of other names, e.g. KeypointLabeler.
So I changed it to this:

Renamed all appearances of the term "KeyPoint" within types and names to "Keypoint".

**Comment by mkamalza** [comment]
> Who is that @jhogins imposter :D

haha nice one

File: com.unity.perception/CHANGELOG.md [hunk 3]
```diff
@@ -9,12 +9,18 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Upgrade Notes
 
+All appearances of the term `KeyPoint` have been renamed to `Keypoint`. Therefore, the following steps will be required after upgrading to this version:
+* If you have code that relies on any renamed types or names, alter your code to reflect the new names.
+* If you are using custom `KeypointTemplate` assets besides the provided `CocoKeypointTemplate`, open those assets in a text editor and change the `keyPoints:` line to `keypoints:`, then save the file.
```
Comment (JonathanHUnity) [inline com.unity.perception/CHANGELOG.md:14]: Can you use `[FormerlySerializedAs("keyPoints")]` to get this automatically?

**Review by JonathanHUnity — CHANGES_REQUESTED** [review]
One last thing to look into.

File: com.unity.perception/CHANGELOG.md (hunk 3 above)
Comment (mkamalza) [reply com.unity.perception/CHANGELOG.md:14]: yes! will do.

**Review by JonathanHUnity — APPROVED** [review]
